### PR TITLE
(FM-8355) Add spec_helper_acceptance

### PR DESCRIPTION
**Pull request by florindragos**
This way, we just need to `require 'litmus_spec_helper_acceptance'` in the spec_helper_acceptance of every module

### CODE REVIEW

**Comment by ekohl** [comment]
Related issue: https://github.com/puppetlabs/puppet_litmus/issues/48

**Comment by florindragos** [comment]
> If you wrap all of this in a module function (e.g.`PuppetLitmus.configure!`) then you could write some unit tests around its behavior. Further, you could make the method take an options hash or yield a configuration object to a block to allow for more explicit customization, even if the defaults continue to look up things from the `ENV` and node config files.
> 
> Usage in `spec_helper_acceptance.rb` then looks like:
> 
> ```ruby
> require 'puppet_litmus'
> PuppetLitmus.configure!
> ```
> 
> or even:
> 
> ```ruby
> begin
>   require 'puppet_litmus'
>   PuppetLitmus.configure!
> rescue LoadError
>   puts 'Unable to load Litmus, did you add it to your Gemfile?'
> end
> ```

We would still need to do `include PuppetLitmus` to have all the helper methods available for the specs unless there is some ruby magic to include a module by calling one of its instance methods.

**Comment by florindragos** [comment]
Updated code works with
```ruby
require 'puppet_litmus'

include PuppetLitmus
PuppetLitmus.configure!
```

**Comment by ekohl** [comment]
Looks like there's a merge conflict so a rebase is needed.

File: lib/litmus_spec_helper_acceptance.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          endpoint: endpoint,
+          operation_timeout: 300,
+        }
+
+        winrm = WinRM::Connection.new opts
+        Specinfra.configuration.winrm = winrm
+      end
+    end
+  end
+end
```
Comment (ekohl) [inline lib/litmus_spec_helper_acceptance.rb]: I'm a big fan of newlines

File: lib/puppet_litmus.rb [hunk 2]
```diff
@@ -7,6 +7,7 @@ module PuppetLitmus; end
 require 'puppet_litmus/inventory_manipulation'
 require 'puppet_litmus/puppet_helpers'
 require 'puppet_litmus/rake_helper'
+require 'litmus_spec_helper_acceptance'
```
Comment (ekohl) [inline lib/puppet_litmus.rb]: I think this should be in `puppet_litmus` as well:
```suggestion
require 'puppet_litmus/spec_helper'
```

I'm also wondering if it should be included by default.

Comment (florindragos) [reply lib/puppet_litmus.rb]: yeah, that would make more sense

File: lib/litmus_spec_helper_acceptance.rb (hunk 1 above)
Comment (florindragos) [reply lib/litmus_spec_helper_acceptance.rb]: so is rubocop :)

Comment (florindragos) [reply lib/litmus_spec_helper_acceptance.rb]: for some reason the newline does not show up here...

Comment (ekohl) [reply lib/litmus_spec_helper_acceptance.rb]: It's there. There is a special symbol Github places here if there's none.

**Comment by ekohl** [comment]
With the last upgrade the `include PuppetLitmus` is no longer needed, correct?

**Comment by florindragos** [comment]
> With the last upgrade the `include PuppetLitmus` is no longer needed, correct?

yep, just tested

```require 'puppet_litmus'
PuppetLitmus.configure!```

**Pull request merged**